Week 1 663380576-0 นายศุภชัย คนเพียร

Lab 1.1: Environment Setup & First Script

In [1]:
print("Hello World")

Hello World


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [2]:
Age = 22
GPA = 2.13
Favorite_Color = "Blue"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  22 <class 'int'>
GPA =  2.13 <class 'float'>
Favorite_Color =  Blue <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [3]:
num1 = 7
num2 = 6
sum = num1 * num2
print("Sum = ",sum)

Sum =  42


4. String concatenation and f-strings:


In [4]:
First_name = "Suphachai"
Last_name = "khonphian"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Suphachai khonphian


5. User input and type conversion:


In [5]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What your favorite number : 80
type of A is :  <class 'str'>
80  + 5 =  85


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: โปรแกรมเก็บข้อมูลชนิดสินค้า,โปรแกรมคำนวณเลขต่างๆ,เขียนเว็บไซต์,สร้างเกมต่างๆ

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [14]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: 13.5
<class 'float'>

ทำไมถึงเป็นtype float เพราะ หาก int ถูกบวกกับ float จะเป็น float เองอัตโนมัติ

In [12]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old.
<class 'str'>

ทำไมถึงเป็นtype str เพราะว่า ตัวแปร message เป็นเก็บข้อมูลเป็นตัวอักษร(str)

In [10]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: False
<class 'bool'>

ทำไมถึงเป็นtype bool เพราะว่า ข้อมูลเก็บได้แค่ True,False

In [ ]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ได้ผลลัพธ์: Error

ทำไมถึงเป็นtype ตัวแปรไม่ตรงกัน ตัวแปรข้อมูลของ num_str(str) เป็นตัวอักษร และ result(int) เป็นตัวเลขต้องแปรเป็น str ถึงจะ ทำให้สามารถเข้ากัน

In [16]:
# Puzzler 4
# What happens here?
num_str = input("Enter a number: ")
result = int(num_str) + 5
print(result)

Enter a number: 10
15


ได้ผลลัพธ์: Enter a number: 10
15

ทำไมถึงเป็นtype int เพราะว่า num_str ถูกกำหนดเป็น int และบวก 5(int) ทำให้ตัวแปร result เป็น int